### ISSUE TITLE
Uniform model settings introspection (`model.settings` / `get_params()`)

### ISSUE THREAD

**Issue by nealcaren** [nealcaren/topica#400]
Surfaced while building the analysis manifest (#394): there is no uniform way to read a fitted model's constructor configuration back off the object.

### Problem
Models expose their settings unevenly — `LDA` has `alpha`/`beta`, `STM` exposes `num_topics` but not much before fitting, and most models expose nothing structured. To record settings, the manifest maintains a hand-written per-model allowlist (`_SETTINGS_ALLOWLIST`, currently LDA + STM only) and falls back to "generic" for everything else. That does not scale to ~40 models and is a second inventory to keep in sync.

### Proposal
Add a uniform, sklearn-style introspection surface returning the model's configuration as a JSON-serialisable dict:

```python
model.settings          # or model.get_params()
# -> {"num_topics": 20, "alpha": [...], "beta": 0.01, "seed": 42, ...}
```

- Cover every model with its own hyperparameters (the constructor arguments), keyword-named to match the constructor.
- Include `seed` (see #399), so a single call captures the full construction config.

### Why
- **Generally useful**: anyone introspecting or logging a model wants this; it is a standard estimator affordance.
- **Retires a manifest V2 backlog item**: with `model.settings` in place, the manifest just calls it and the per-model settings adapters (the deferred V2 "registry adapters for every model family") are no longer needed — the fix belongs on the model, not in the manifest.

Relates to #394 (and would replace the manifest's `_SETTINGS_ALLOWLIST`).

**Comment by nealcaren** [comment]
Independent second opinion (OpenAI Codex, focused review) — endorses the uniform surface and sharpens the implementation:

**Verdict: yes, add it.** Use sklearn's actual contract — `get_params()` returning constructor parameters as a dict keyed by `__init__` argument names (models can ignore `deep`), with an optional read-only `settings` alias. Familiar for provenance, cloning, and inspection.

Key refinements:
- **Do not derive settings from Rust internal fields** — internal representation often differs from user-facing configuration. Instead declare each model's constructor config **once**, via a declarative Rust macro/trait that generates `#[new]`, `get_params`/`settings`, and (optionally) the property getters from one spec. Each pyclass supplies values to that spec rather than maintaining a second allowlist — that is what actually kills the drift.
- `settings` should be a snapshot of **explicit construction config**: JSON-safe, deterministic (primitives/strings/lists/dicts, normalized enums), `seed` included consistently, and **excluding** fitted state, derived caches, and corpus-dependent dimensions.

Risks flagged:
- **Schema freezing** — once emitted in provenance, renaming keys or changing default semantics becomes a compatibility commitment. **Version the settings schema.**
- **Mutable/reference values** — return copies / normalize arrays and maps, or callers can mutate the model's apparent configuration.
- **Constructor ambiguity** — decide whether "user omitted a value" vs "a default was used" needs to be distinguishable; otherwise report effective settings consistently.

Net: this supersedes the manifest's `_SETTINGS_ALLOWLIST` — the manifest would just call `get_params()`.
